doc/rtd/development/datasource_creation.rst

```diff
@@ -86,25 +88,43 @@ There are multiple ways to provide `user data`, `metadata`, and
 `vendor data`, and each cloud solution prefers its own way. A datasource
 abstract base class defines a single interface to interact with the different
 clouds. Each cloud implementation must inherit from this base class to use this
-shared functionality and interface. See :file:`cloud-init/sources/__init__.py`
+shared functionality and interface. See `cloudinit/sources/__init__.py`_
 to see this class.
 
 If you are interested in adding a new datasource for your cloud platform you
 will need to do all of the following:
 
+Update ``ds-identify``
+----------------------
+
+In ``systemd``, Alpine and BSD environments, ``ds-identify`` runs in early boot
+to detect which datasource should be enabled, or if ``cloud-init`` should run
+at all. You'll need to add early identification support for the platform via a
+``dscheck_<CloudPlatform>`` function in `tools/ds-identify`_.
+For example, see `NWCS support`_
+
+Add tests for ``ds-identify``
+-----------------------------
+
+Add relevant tests in a new class to
+`tests/unittests/test_ds_identify.py`_. Use ``TestOracle`` as an example.
+
 Add datasource module cloudinit/sources/DataSource<CloudPlatform>.py
 --------------------------------------------------------------------
 
-We suggest you start by copying one of the simpler datasources
-such as ``DataSourceHetzner``.
+Use one of the simpler datasources such as ``DataSourceHetzner`` as a guiding
+template for style and expectations. The DataSource module should implement a
+``ds_detect`` method validates the same identification conditions defined
+in ds-identify and returns ``True`` when met. This allows cloud-init to support
```

Comment:
@blackboxsw add netv2 discussion here. from Alberto

Reply:
I do not see changes covering this. Maybe they got mangled by some git operation?